dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/TimeFieldSqlRenderer.java

```diff
@@ -214,4 +229,40 @@ protected abstract String getColumnName(
    * @return the collection of {@link TimeField}
    */
   protected abstract Set<TimeField> getAllowedTimeFields();
+
+  protected String getSqlForAllPeriods(String alias, List<DimensionalItemObject> periods) {
```

Comment:
Would be nice to have a Javadoc here. Maybe showing some examples of possible SQL snippets output.

Reply:
done!